### PR TITLE
fix: update parser-js to v2

### PR DESCRIPTION
**Pull request by smoya**
Replaces https://github.com/asyncapi/studio/pull/572

**Description**

This PR updates **@asyncapi/parser** to v2 and uses the recently released custom schema parsers.

Note: this PR does not change Studio to start using the new Parser-API but keeps using the old wrapper `const oldDocument = convertToOldAPI(document);`.

**Related issue(s)**
https://github.com/asyncapi/parser-js/issues/585

### CODE REVIEW

File: src/services/parser.service.ts [hunk 1]
```diff
@@ -28,8 +28,8 @@ export class ParserService extends AbstractService {
     });
 
     this.subscribeToFiles();
-    this.subscribeToSettings();
-    this.parseSavedDocuments();
+    await this.subscribeToSettings();
+    await this.parseSavedDocuments();
```
Comment (magicmatatjahu) [inline src/services/parser.service.ts:32]: It won't wait because subscription is a procedure, not function :) So I don't think so that this is needed.

File: src/index.tsx [hunk 2]
```diff
@@ -32,7 +32,7 @@ function configureMonacoEnvironment() {
   };
 }
 
-async function bootstrap() {
+(async () => {
```
Comment (magicmatatjahu) [inline src/index.tsx:35]: Why did you change `bootstrap` function to something like that?

File: src/services/parser.service.ts (hunk 1 above)
Comment (smoya) [reply src/services/parser.service.ts:32]: SonarCloud complains about this being missing.

File: src/index.tsx (hunk 2 above)
Comment (smoya) [reply src/index.tsx:35]: SonarCloud complains (marks as a bug) with something like this iirc: "Promises must be awaited, end with a call to .catch, or end with a call to .then with a rejection handler"

Comment (magicmatatjahu) [reply src/index.tsx:35]: As we see, Sonarcloud was written by Java people 😆 Ok.

**Comment by smoya** [comment]
> Something doesn't work. When I remove `info.title` from the spec then I see errors, but when I update the config to hide the warnings they aren't filtered (still with errors). Rest are good for me. Please only check that described bug, because it works as I described in current studio https://studio.asyncapi.com/

Being honest, I don't understand why changing the following lines back to what it was previously fixes the issue: https://github.com/asyncapi/studio/pull/615/commits/768b01c3aa0f8faab0498e813f566b824f53fe3e
